### PR TITLE
Refine Event Handling on Home Screen

### PR DESCRIPTION
**Pull request by Nath-Assa**
### Key Changes:

- Bug Fix: Previously, navigating away from and then returning to the Upcoming tab incorrectly displayed all events instead of just the upcoming ones. This issue has been resolved.

- When you have no upcomings events, pressing on map icon now shows the map.

- Enhanced the EventsOverviewUiState by introducing a new upcoming events list to differentiate between all events and specifically upcoming ones. This update ensures clearer state management and optimizes UI updates specific to upcoming events, improving the responsiveness and accuracy of event displays.

### CODE REVIEW

**Review by christinehahaa — APPROVED** [review]
Great - these changes will be useful for reducing code complexity and the search and filter functionality as well. I have also tested on an android. LGTM!

File: app/src/main/java/com/github/se/eventradar/ui/home/Home.kt [hunk 1]
```diff
@@ -183,7 +188,7 @@ fun HomeScreen(
                       start.linkTo(parent.start)
                       end.linkTo(parent.end)
                     })
-        (uiState.eventList.allEvents.isEmpty()) ->
+        (uiState.viewList && uiState.upcomingEventList.allEvents.isEmpty()) ->
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/home/Home.kt:191]: this is not a very good use of the when block. This logic should be split. One if/when statement for dealing with whether the list or the map is shown, and another for what will be shown based on the uiState. Having them together like this is not very readable.

**Review by jgeaso1266 — COMMENTED** [review]
One comment, otherwise looks good.

File: app/src/main/java/com/github/se/eventradar/ui/home/Home.kt (hunk 1 above)
Comment (Nath-Assa) [reply app/src/main/java/com/github/se/eventradar/ui/home/Home.kt]: done

Comment (Nath-Assa) [reply app/src/main/java/com/github/se/eventradar/ui/home/Home.kt]: After checking Sonar Cloud, this change actually makes our cognitive complexity higher. Reverting the changes.

**Comment by Nath-Assa** [comment]
Closes #220

**Pull request merged**